src/core/Pagination/PageInput/PageInput.tsx

```diff
@@ -0,0 +1,278 @@
+import { getConditionalAriaProp } from '../../../utils/aria';
+import {
+  HtmlInput,
+  HtmlInputProps,
+  HtmlSpan,
+  HtmlDiv,
+  HtmlDivProps,
+  HtmlButton,
+  HtmlButtonProps,
+} from '../../../reset';
+import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
+import { StatusText } from '../../Form/StatusText/StatusText';
+import { Label } from '../../Form/Label/Label';
+import { Icon } from '../../Icon/Icon';
+import { InputStatus, StatusTextCommonProps } from '../../Form/types';
+import { baseStyles } from './PageInput.baseStyles';
+
+type PageInputValue = number | string | null;
+
+type PageInputStatus = Exclude<InputStatus, 'success'>;
+
+export interface PageInputProps
+  extends StatusTextCommonProps,
+    Omit<
+      HtmlInputProps,
+      | 'type'
+      | 'disabled'
+      | 'onChange'
+      | 'children'
+      | 'onClick'
+      | 'value'
+      | 'defaultValue'
+    > {
+  /** PageInput container div class name for custom styling. */
+  className?: string;
+  /** PageInput wrapping div element props */
+  wrapperProps?: Omit<HtmlDivProps, 'className'>;
+  /** Label text */
+  labelText: ReactNode;
+  /** Placeholder text for input. */
+  visualPlaceholder?: string;
+  /** Page button label for screen readers */
+  pageInputButtonLabel: string;
+  /** PageButtonProps */
+  pageInputButtonProps?: Omit<HtmlButtonProps, 'onClick' | 'tabIndex'>;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: PageInputStatus;
+  /** Input name */
+  name?: string;
+  /** Set components width to 100% */
+  fullWidth?: boolean;
+  /** Callback for onBlur event */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /** Callback for search button click */
+  onPageChange?: (value: PageInputValue) => void;
+  /** Maximum value */
+  maxValue: number;
+}
+
+const baseClassName = 'fi-page-input';
+const pageInputClassNames = {
+  fullWidth: `${baseClassName}--full-width`,
+  error: `${baseClassName}--error`,
+  notEmpty: `${baseClassName}--not-empty`,
+  labelIsVisible: `${baseClassName}_label--visible`,
+  styleWrapper: `${baseClassName}_wrapper`,
+  inputElement: `${baseClassName}_input`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  functionalityContainer: `${baseClassName}_functionality-container`,
+  statusTextHasContent: `${baseClassName}_statusText--has-content`,
+  button: `${baseClassName}_button`,
+  pageInputIcon: `${baseClassName}_button-icon`,
+};
+
+interface PageInputState {
+  value: PageInputValue;
+  status: PageInputStatus;
+  inputValue: string | number | undefined;
+}
+
+class BasePageInput extends Component<PageInputProps & SuomifiThemeProp> {
+  state: PageInputState = {
+    value: '',
+    status: this.props.status || 'default',
+    inputValue: undefined,
+  };
+
+  private inputRef = createRef<HTMLInputElement>();
+
+  private getStatusText = () =>
+    `"${this.state.inputValue}" ${this.props.statusText}`;
```

Comment:
Should the statusText be a function that the developer can pass to allow more flexibility?

Reply:
Or simply leave this logic out and let the developer get the value and add it to the statusText if they want.

Reply:
There is now a function prop to handle this